src/pybind/mgr/cephadm/serve.py

```diff
@@ -1375,6 +1377,32 @@ def _to_args(lst: ArgumentList) -> List[str]:
             eea = None
         return daemon_spec, eca, eea
 
+    def _setup_init_containers(
+        self,
+        daemon_spec: CephadmDaemonDeploySpec,
+        params: Dict[str, Any],
+    ) -> List[Dict[str, Any]]:
+        """Handle any init containers - containers that run before a daemon (detached)
+        container that are expected to run for a short time and then exit.
+        """
+        # the expanded_init_containers function allows a type to customize
+        # or otherwise generate init_containers without needing to store
+        # a full list of init containers in an attribute
+        expanded = getattr(daemon_spec, 'expanded_init_containers', None)
+        if expanded:
+            ics = expanded()
+        else:
+            # the simpler more direct form of init containers
+            ics = getattr(daemon_spec, 'init_containers', None)
+        if not ics:
+            return []
+        ic_meta = []
+        ic_params = params['init_containers'] = []
+        for ic in ics:
+            ic_meta.append(ic.to_json())
+            ic_params.append(ic.to_json(flatten_args=True))
+        return ic_meta
```

Comment:
I'm confused why we're building this `ic_params` here when we don't end up returning it.

Reply:
It's an alias `ic_params = params['init_containers'] = []` that side effects the params dict passed in. This is similar to the behavior of  _setup_extra_deployment_args as I was intentionally mimicking that function.  Do you think the aliasing is too confusing?